mesonbuild/environment.py

```diff
@@ -134,18 +134,28 @@ def detect_gcovr(min_version='3.3', new_rootdir_version='4.2', log=False):
         return gcovr_exe, mesonlib.version_compare(found, '>=' + new_rootdir_version)
     return None, None
 
+def detect_llvm_cov():
+    tools = get_llvm_tool_names('llvm-cov')
+    for tool in tools:
+        if mesonlib.exe_exists([tool, '--version']):
+            return tool
+
 def find_coverage_tools():
     gcovr_exe, gcovr_new_rootdir = detect_gcovr()
 
+    llvm_cov_exe = detect_llvm_cov()
+
     lcov_exe = 'lcov'
     genhtml_exe = 'genhtml'
 
     if not mesonlib.exe_exists([lcov_exe, '--version']):
         lcov_exe = None
     if not mesonlib.exe_exists([genhtml_exe, '--version']):
         genhtml_exe = None
+    if not mesonlib.exe_exists([llvm_cov_exe, '--version']):
+        llvm_cov_exe = None
```

Comment:
You don't need to call `exe_exists her again, you've already done that in `detect_llvm_cov`.